util/topsql/tracecpu/profile.go

```diff
@@ -62,123 +51,90 @@ type SQLCPUTimeRecord struct {
 	CPUTimeMs  uint32
 }
 
-type sqlCPUProfiler struct {
-	taskCh chan *profileData
+// SQLCPUCollector uses to consume cpu profile from globalCPUProfiler, then parse the SQL CPU usage from the cpu profile data.
+type SQLCPUCollector struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
 
-	mu struct {
-		sync.Mutex
-		ept *exportProfileTask
-	}
-	collector atomic.Value
+	profileConsumer cpuprofile.ProfileConsumer
+	collector       Collector
 }
 
-var (
-	defaultProfileBufSize = 100 * 1024
-	profileBufPool        = sync.Pool{
-		New: func() interface{} {
-			return bytes.NewBuffer(make([]byte, 0, defaultProfileBufSize))
-		},
-	}
-)
-
-// newSQLCPUProfiler create a sqlCPUProfiler.
-func newSQLCPUProfiler() *sqlCPUProfiler {
-	return &sqlCPUProfiler{
-		taskCh: make(chan *profileData, 128),
+// NewSQLCPUCollector create a SQLCPUCollector.
+func NewSQLCPUCollector(c Collector) *SQLCPUCollector {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &SQLCPUCollector{
+		ctx:             ctx,
+		cancel:          cancel,
+		profileConsumer: make(cpuprofile.ProfileConsumer, 1),
+		collector:       c,
 	}
 }
 
-func (sp *sqlCPUProfiler) Run() {
-	logutil.BgLogger().Info("cpu profiler started")
-	go sp.startCPUProfileWorker()
+// Start uses to start to run SQLCPUCollector.
+func (sp *SQLCPUCollector) Start() {
+	logutil.BgLogger().Info("sql cpu collector started")
+	sp.wg.Add(1)
 	go sp.startAnalyzeProfileWorker()
 }
 
-func (sp *sqlCPUProfiler) SetCollector(c Collector) {
-	sp.collector.Store(c)
-}
-
-func (sp *sqlCPUProfiler) GetCollector() Collector {
-	c, ok := sp.collector.Load().(Collector)
-	if !ok || c == nil {
-		return nil
+// Close uses to close the SQLCPUCollector.
+func (sp *SQLCPUCollector) Close() {
+	if sp.cancel != nil {
+		sp.cancel()
 	}
-	return c
+	sp.wg.Wait()
 }
 
-func (sp *sqlCPUProfiler) startCPUProfileWorker() {
-	defer util.Recover("top-sql", "profileWorker", nil, false)
-	for {
-		if sp.IsEnabled() {
-			sp.doCPUProfile()
-		} else {
-			time.Sleep(time.Second)
-		}
-	}
+// Enable uses to enable the SQLCPUCollector to work.
```

Comment:
This API design is very uncommon -- in your API design, a `Start` does not make it functional, but a `Enable` will make it functional. Then what does `Start` do? It is usually a signal of bad API design, as a good API design should be fully understandable and reasonable according to the public API (w/ API doc comments).